### PR TITLE
add testcases to test auth/oauth2.py

### PR DESCRIPTION
**Pull request by muryanto1**
@jasonb5  Please take a look at my changes to auth/oauth2.py.  I wonder if refresh_certificate() and check_certificated() ever get exercised in production.  
Also, I tried to add the setting of these environment variables in the setUp() code:
    def setUp(self):
        os.environ['OAUTH2_CLIENT_ID'] = ''
        os.environ['OAUTH2_CLIENT_SECRET'] = ''
        os.environ['SECRET_KEY'] = 'some_secret_key'
        os.environ['PROVISIONER_FRONTEND'] = '127.0.0.1'
        self.user = models.User.objects.first()

but it did not help, the environment variables were still not set during test runs.

This posting https://stackoverflow.com/questions/31195183/django-setting-environment-variables-in-unittest-tests suggests to use "override_settings", but it needs changes to Django's settings.py

### CODE REVIEW

File: compute/compute_wps/compute_wps/tests/test_oauth2.py [hunk 1]
```diff
@@ -3,18 +3,51 @@
...
+                                                    'test_email@test.com', 'test_password1')
+        auth = models.Auth.objects.create(openid_url=self.openid_url, user=test_user)
+        auth.update(auth_type, self.certs, self.api_key, token=self.test_token, state='good_state')
+        test_token, test_state = test_user.auth.get('token', 'state')
+        return auth, test_user
 
     def setUp(self):
-        os.environ['OAUTH_CLIENT'] = ''
-        os.environ['OAUTH_SECRET'] = ''
+        os.environ['OAUTH2_CLIENT_ID'] = ''
```
Comment (jasonb5) [inline compute/compute_wps/compute_wps/tests/test_oauth2.py:47]: Yea this won't work, you'll need to use mock.patch.dict as a method decorator. See https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict.

**Comment by jasonb5** [comment]
The methods refresh_certificate() and check_certificated() are called here https://github.com/ESGF/esgf-compute-wps/blob/a2f1f3ddfeff21378da6fe5ee570a53a9410f5e5/compute/compute_wps/compute_wps/views/esgf.py#L52-L53

File: compute/compute_wps/compute_wps/tests/test_oauth2.py (hunk 1 above)
Comment (muryanto1) [reply compute/compute_wps/compute_wps/tests/test_oauth2.py:47]: I tried:

@mock.patch.dict('os.environ', {'OAUTH2_CLIENT_ID': "",
                                    'DUMMY_VAR': "DUMMY_VAL",
                                    'OAUTH2_CLIENT_SECRET': "",
                                    'PROVISIONER_FRONTEND': "127.0.0.1",
                                    'SECRET_KEY': "hmmmm"})
    def setUp(self):
        print("...setUp...DUMMY_VAR: ", os.environ['DUMMY_VAR'])
        print("...setUp...OAUTH2_CLIENT_ID: ", os.environ['OAUTH2_CLIENT_ID'])
        print("...setUp...OAUTH2_CLIENT_SECRET: ", os.environ['OAUTH2_CLIENT_SECRET'])
        print("...setUp...PROVISIONER_FRONTEND: ", os.environ['PROVISIONER_FRONTEND'])
        print("...setUp...SECRET_KEY: ", os.environ['SECRET_KEY'])
        self.user = models.User.objects.first()

but when I ran the code:

(base) root@3f26133a864f:/src/compute_wps# python /compute/manage.py test compute_wps.tests.test_oauth2
Traceback (most recent call last):
  File "/compute/manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/conda/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/opt/conda/lib/python3.6/site-packages/django/core/management/__init__.py", line 308, in execute
    settings.INSTALLED_APPS
  File "/opt/conda/lib/python3.6/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/opt/conda/lib/python3.6/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/opt/conda/lib/python3.6/site-packages/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/opt/conda/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/compute/compute/settings.py", line 70, in <module>
    SECRET_KEY = os.environ['SECRET_KEY']
  File "/opt/conda/lib/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'SECRET_KEY'


I could be wrong, but it seems to be that the environment variables are accessed before the test run gets to the test code.

**Comment by jasonb5** [comment]
@muryanto1 We discussed not adding any additional testing can i go ahead and merge this? i need the utc fix.

**Comment by muryanto1** [comment]
> @muryanto1 We discussed not adding any additional testing can i go ahead and merge this? i need the utc fix.

@jasonb5 yes, sure.

**Review by jasonb5 — APPROVED** [review]

**Pull request merged**